ui.frontend/src/main/webpack/site/js/components/loadMoreButton.ts

```diff
@@ -0,0 +1,58 @@
+import { OnLoadMoreButtonClickCallback } from '../types/callbacks'
+import { buildSearchItem, SearchItem } from './searchItem'
+
+type SearchButtonOptions = {
+  loadMoreButtonText: string
+  offset: number
+  tabUrl: string
+  searchValue: string
+  searchResultsElement: HTMLDivElement
+  onLoadMoreButtonClick?: OnLoadMoreButtonClickCallback
+}
+
+const buildLoadMoreButton = ({
+  loadMoreButtonText,
+  offset,
+  tabUrl,
+  searchValue,
+  searchResultsElement,
+  onLoadMoreButtonClick,
+}: SearchButtonOptions): HTMLButtonElement => {
+  const loadMoreButton = document.createElement('button')
+  loadMoreButton.classList.add('saas-load-more-button')
+
+  loadMoreButton.dataset.offset = `${offset}`
+
+  loadMoreButton.innerText = loadMoreButtonText
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  loadMoreButton.addEventListener('click', async (event) => {
+    onLoadMoreButtonClick?.()
+
+    event.preventDefault()
+
+    const currentOffset = loadMoreButton.dataset.offset || offset
```

Comment:
Probably i've communicated the meaning of "start" param's value incorrectly.
For paginations there are two params used in saas api ("start" and "rows").
start - the result page index in the pagination
rows - the results per page
example 1: start = 0; rows = 10 - will fetch the first 10 result.
example 2: start = 2; rows = 10 - will fetch the results from 21 - 30.
I decided for the AEM search api to use the same param name "start", and rows (configurable in search cmp's dialog) will be applied on BE side. So for the load more the value of start should increase by 1.

Reply:
Fixed. I adapted the offset value when invoking `buildLoadMoreButton` in `searchForm.ts`